app/(main)/account/recovery/page.tsx

```diff
@@ -0,0 +1,261 @@
+import Alert from '@/components/AlertNotification/AlertNotification';
+import { AlertVariants } from '@/components/AlertNotification/Alerts.enum';
+import { Button } from '@/components/Button/Button';
+import { Input } from '@/components/Input/Input';
+import LoadingSpinner from '@/components/LoadingSpinner/LoadingSpinner';
+import Logo from '@/components/Logo/Logo';
+import { useAuthContext } from '@/context/AuthContextProvider';
+import logo from '@/public/assets/logo-colored-outline.svg';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useRouter, useSearchParams } from 'next/navigation';
+import React, { useEffect, useState } from 'react';
+import { Control, SubmitHandler, useForm, useWatch } from 'react-hook-form';
+import toast from 'react-hot-toast';
+import { z } from 'zod';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from '../../../../components/Form/Form';
+import { Suspense } from 'react';
+import GlobalSpinner from '@/components/GlobalSpinner/GlobalSpinner';
+
+interface CustomError extends Error {
+  type?: string;
+}
+
+/**
+ * The schema for the login form.
+ * @throws {Error} Throws an error if the email is not a valid email address.
+ * @throws {Error} Throws an error if the email is not provided.
+ * @throws {Error} Throws an error if the email is not a string.
+ */
+const ResetPasswordSchema = z
+  .object({
+    password: z
+      .string()
+      .min(1, { message: 'Please enter a password' })
+      .min(6, { message: 'Password must be at least 6 characters' }),
+    confirmPassword: z
+      .string()
+      .min(1, { message: 'Please confirm your password' })
+      .min(6, { message: 'Password must be at least 6 characters' }),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    message: "Passwords don't match",
+    path: ['confirmPassword'],
+  });
+
+type ResetUserPasswordSchemaType = z.infer<typeof ResetPasswordSchema>;
+
+/**
+ * Renders the recover password page.
+ * @returns {React.JSX.Element} The rendered login page.
+ */
+const ResetPassword = (): React.JSX.Element => {
+  const router = useRouter();
+  const { isSignedIn, getUser } = useAuthContext();
+  const [isLoading, setIsLoading] = useState(true);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  let userId: string | null = null;
+  let secret: string | null = null;
+  let expire: string | null = null;
+
+  /**
+   * Wrapper component to handle search params
+   * @returns {null} - null
+   */
+  const SearchParamsWrapper = (): null => {
+    const searchParams = useSearchParams();
+    userId = searchParams.get('userId');
+    secret = searchParams.get('secret');
+    expire = searchParams.get('expire');
+
+    useEffect(() => {
+      if (!userId || !secret || !expire) {
+        router.push('/login');
+      } else {
+        setIsLoading(false);
+        if (isSignedIn) {
+          getUser();
+          router.push('/league/all');
+        }
+      }
```

Comment:
Do we need the else here? Could we not add a `return` in the if statement?

Reply:
I think this is a nit.

Reply:
nit: remove `else` and add a `return` in `if` statement

Reply:
✅ fixed.